app/controllers/incomeSources/manage/ConfirmReportingMethodSharedController.scala

```diff
@@ -185,18 +204,26 @@ class ConfirmReportingMethodSharedController @Inject()(val manageIncomeSources:
                               isAgent: Boolean,
                               successCall: Call,
                               taxYears: TaxYear,
-                              incomeSourceId: String,
+                              incomeSourceIdMaybe: Option[String],
                               reportingMethod: String
                              )(implicit user: MtdItUser[_], hc: HeaderCarrier): Future[Result] = {
 
-    updateIncomeSourceService.updateTaxYearSpecific(
-      nino = user.nino,
-      incomeSourceId = incomeSourceId,
-      taxYearSpecific = TaxYearSpecific(taxYears.endYear.toString, reportingMethod match {
-        case "annual" => true
-        case "quarterly" => false
-      })
-    ) flatMap {
+
+    val updateIncomeSourceResFuture = for {
+      updateIncomeSourceRes <- incomeSourceIdMaybe match {
+        case Some(incomeSourceId) => updateIncomeSourceService.updateTaxYearSpecific(
+          nino = user.nino,
+          incomeSourceId = incomeSourceId,
+          taxYearSpecific = TaxYearSpecific(taxYears.endYear.toString, reportingMethod match {
```

Comment:
this is really optional at this stage, but we would need to wrap this down to the type level  (TaxYearSpecific )